lib/mixlib/shellout/windows.rb

```diff
@@ -196,6 +196,33 @@ def consume_output(open_streams, stdout_read, stderr_read)
         true
       end
 
+      # Use to support array passing semantics on windows
+      #
+      # 1.  strings with whitespace or quotes in them need quotes around them.
+      # 2.  interior quotes need to get backslash escaped (parser needs to know when it really ends).
+      # 3.  random backlsashes in paths themselves remain untouched.
+      # 4.  if the argument must be quoted by #1 and terminates in a sequence of backslashes then all the backlashes must themselves
+      #     be backslash excaped (doubling the backslashes)
+      # 5.  if an interior quote that must be escaped by #2 has a sequence of backslashes before it then all the backslashes must
+      #     themself be backslash excaped along with the backslash ecape of the interior quote
+      # see: https://blogs.msdn.microsoft.com/twistylittlepassagesallalike/2011/04/23/everyone-quotes-command-line-arguments-the-wrong-way/
+      #
+      # @api private
+      # @param args [Array<String>] array of command arguments
+      # @return String
+      def combine_args(*args)
+        return args[0] if args.length == 1
+        args.map do |arg|
+          if arg =~ /[ \t\n\v"]/
+            arg = arg.gsub(/(\\*)"/, '\\1\\1\"') # fix interior quotes and any preceding backslashes
+            arg = arg.sub(/(\\+)$/, '\\1\\1') # handle terminating backslashes before the quotes
```

Comment:
This part confuses me a bit. Normally, it is something like `str.gsub(/pa(tt)ern/, '\1')`... does this have the same effect?

Now I'm EXTRA confused:

```ruby
>> str = 'blah\\\\"'
>> .gsub(/(\\*)"/, '\1\1"')
=> "blah\\\\\\\\\""                #  9 backslashes
>> str.gsub(/(\\*)"/, '\\1\\1"')
=> "blah\\\\\\\\\""                #  9 backslashes
>> str.gsub(/(\\*)"/, '\\1\\1\"')
=> "blah\\\\\\\\\\\""              # 11 backslashes
```

So there is no difference between your double-backslash-one vs my single-backslash-one, but that backslash-doublequote totally effs it up and adds another. Can you double-check that that is what you actually want?

Reply:
okay, the double backslashes are just some kind of mental copypasta error, removing them does nothing to the tests and the capture group works fine.

and i think the quotes are correct.  because the argument in the gsub is single-quoted \" is not an escape sequence so that is two-characters a real backslash and a real quote -- which is intended to have the backslash quote the quote character.

in the double quoted return string when you see 11 backslashes that is 5 actual backslashes which have to be escaped by ruby and the embedded quote character which also has to be backslashed by ruby.

that is the actual algorithm which is that two (real) backslashes before an embedded quote must be converted into 2N+1 real backslashes plus the quote.  so 5 backslashes.  which is 11 backslashes to keep ruby's double-quotes happy.

fun.